cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java

```diff
@@ -128,9 +129,13 @@ public static void createBaloInHomeRepo(HttpURLConnection conn, Path pkgPathInBa
      *
      * @param pkgVersion   package version
      * @param logFormatter log formatter
+     * @return valid package version
      */
-    static void validatePackageVersion(String pkgVersion, LogFormatter logFormatter) throws CentralClientException {
-        if (!pkgVersion.matches(VERSION_REGEX)) {
+    static String validatePackageVersion(String pkgVersion, LogFormatter logFormatter) throws CentralClientException {
+        try {
+            Version version = Version.valueOf(pkgVersion);
+            return version.toString();
+        } catch (IllegalArgumentException | ParseException e) {
             throw new CentralClientException(logFormatter.formatLog("package version could not be detected"));
```

Comment:
We need to improve this error message. Please have a look at the SemanticVersion class

Reply:
Pls check now

Reply:
Looks good